src/components/CustomTable/CustomTable.tsx

```diff
@@ -0,0 +1,146 @@
+    } else if (selectedIndex === 0) {
+      newSelected = newSelected.concat(selected.slice(1));
+    } else if (selectedIndex === selected.length - 1) {
+      newSelected = newSelected.concat(selected.slice(0, -1));
+    } else if (selectedIndex > 0) {
+      newSelected = newSelected.concat(selected.slice(0, selectedIndex), selected.slice(selectedIndex + 1));
+    }
+
+    setSelected(newSelected);
+  };
+
+  const handleChangePage = (_event: unknown, newPage: number) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event: ChangeEvent<HTMLInputElement>) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  const isSelected = (row: T) => {
+    return (
+      selected.filter((item) => {
+        return item.id === row.id;
+      }).length > 0
+    );
+  };
+
+  const emptyRows = page > 0 ? Math.max(0, (1 + page) * rowsPerPage - rows.length) : 0;
+
+  useEffect(() => {
+    fetch(fetchUrl)
+      .then((res) => {
+        return res.json();
+      })
+      .then((data: T[]) => {
+        return setRows(data);
+      })
+      .catch((error) => {
+        if (error instanceof FetchError) {
+          throw new FetchError(`Failed to fetch users with error: ${error.message}`);
+        }
+      });
+  });
+
+  return (
+    <Box
+      width='100%'
+      height='100%'
+    >
+      {toolbar}
+
+      <TableContainer>
+        <Table>
+          {head}
+          <TableBody>
+            {rows.map((row, index) => {
+              const isItemSelected = isSelected(row);
+              const labelId = `enhanced-table-checkbox-${index}`;
+
+              type TableDataKeys = Exclude<keyof T, symbol>;
+              const columnNames: TableDataKeys[] = Object.keys(row) as TableDataKeys[];
+
+              const tableCells = columnNames.slice(1).map((key) => {
+                return (
+                  <TableCell
+                    key={key}
+                    align='left'
+                  >
+                    {row[key] as ReactNode}
+                  </TableCell>
+                );
+              });
+
+              return (
+                <TableRow
+                  hover
+                  onClick={(event: MouseEvent<unknown>) => {
+                    return handleClick(event, row);
+                  }}
+                  role='checkbox'
+                  aria-checked={isItemSelected}
+                  tabIndex={-1}
+                  key={row.id as number}
+                  selected={isItemSelected}
+                >
+                  <TableCell padding='checkbox'>
+                    <Checkbox
+                      color='primary'
+                      checked={isItemSelected}
+                      inputProps={{
+                        'aria-labelledby': labelId,
+                      }}
+                    />
+                  </TableCell>
+                  {tableCells}
+                </TableRow>
+              );
+            })}
```

Comment:
As discussed, we will implement this a bit more differently for better flexibility + no need to use magic numbers to skip some columns we want to avoid showing